tests/integrationv2/test_session_resumption.py

```diff
@@ -320,3 +320,72 @@ def test_s2nd_falls_back_to_full_connection(managed_process, tmp_path, cipher, c
         assert b'Resumed session' not in results.stdout
         assert to_bytes("Actual protocol version: {}".format(
             s2n_version)) in results.stdout
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", [p for p in PROTOCOLS if p != Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL, S2N], ids=get_parameter_name)
+@pytest.mark.parametrize("other_provider", [S2N], ids=get_parameter_name)
+def test_session_resumption_s2n_client_tls13_server_not_tls13(managed_process, cipher, curve, protocol, provider, other_provider, certificate):
```

Comment:
I'm not sure this scenario warrants an integ test. My first thought was that you could just recreate it using a self talk test. s2n_self_talk_resumption test.c has a lot of complex session resumption tests, you could add it there. You would just need a client with a tls13-capable config and a server with a tls12-capable config.

Reply:
We don't necessarily have to remove the integration test, if it works and doesn't take too long. But I agree that we want self-talk tests.

I can't see the logs for your failing integration tests, but if they're failing on this new test, you'll find that self-talk tests are MUCH easier to debug :)

Reply:
Thanks, I'll add self-talk test. I also cannot see the logs [🤔](https://emojipedia.org/thinking-face/)

Reply:
I added a self-talk test, but it seems to be failing to send the session ticket, I'm getting:

```
s2n_result_is_ok(s2n_test_issue_new_session_ticket(server_conn, client_conn, &no_early_data)) is not true  (/home/ubuntu/s2n-tls/tests/unit/s2n_self_talk_session_resumption_test.c:966)
Error Message: 'a safety check failed'
 Debug String: 'Error encountered in /home/ubuntu/s2n-tls/tests/unit/s2n_self_talk_session_resumption_test.c:135'
```

Reply:
Should be working now, thanks for the help @lrstewart